conans/client/tools/version.py

```diff
@@ -42,9 +46,14 @@ def build(self):
     def __eq__(self, other):
         if not isinstance(other, Version):
             other = Version(other)
-        return self._semver.compare(other._semver) == 0
+        return (self._semver.compare(other._semver) or self._compare_micro(other)) == 0
```

Comment:
This condition is the same from upstream: https://github.com/podhmo/python-semver/compare/0.7.0...0.8.0#diff-61055be4b983e3865cf7d7961b00d2c1R395